### PR TITLE
Improvement of namespace and documentation

### PR DESCRIPTION
**Pull request by dmcglinn**
These changes do not change how the code works only how it is arranged and documented.

### CODE REVIEW

**Comment by dmcglinn** [comment]
I should note that changes in this PR are in response to #90, #88, #87, #86

File: R/mobr.R [hunk 1]
```diff
@@ -1016,197 +956,322 @@ pairwise_t = function(dat_sp, dat_plot, groups, lower_N = NA) {
...
+                    col=NA, log='', leg_loc = 'topleft') {
+    env_data = mob_in$env[ , env_var]
+    grps = unique(env_data)
+    if (is.na(col[1])) 
+        col = rainbow(length(grps))
+    if ('sad' == type) {
+        plot(1, type = "n", xlab = "% abundance (log scale)", ylab = "% species", 
+             xlim = c(0.01, 1), ylim = c(0.01, 1), log = log)
+        for (i in 1:length(grps)) {
+            col_grp = col[i]
```
Comment (rueuntal) [inline R/mobr.R]: I didn't test the function, but could this line potentially trigger the error if the user specify `col` with the wrong length?

File: R/mobr.R [hunk 2]
```diff
@@ -1,51 +1,53 @@
...
     }
-    if (any(colSums(x) == 0)) {
+    if (any(colSums(comm) == 0)) {
         warning("Some species have zero occurrences and will be dropped from the community table")
-        x = x[, colSums(x) != 0]
+        comm = comm[, colSums(comm) != 0]
     }
-    out$comm = data.frame(x)
+    out$comm = data.frame(comm)
     spat_cols = which(names(plot_attr) %in% c('x', 'y'))
```
Comment (dmcglinn) [inline R/mobr.R]: `coords` is still included with `plot_attr` so no change here

File: R/mobr.R [hunk 3]
```diff
@@ -57,290 +59,218 @@ make_comm_obj = function(x, plot_attr, binary=FALSE) {
...
+#' rarefaction(sad, method='indiv', effort=1:10)
+#' # the same results as the above can be generated by supplying the community matrix
+#' rarefaction(inv_comm, method='indiv', effort=1:10)
+#' # sample based rarefaction under random sampling
+#' rarefaction(inv_comm, method='samp')
+#' # sampled based rarefaction under spatially explicit nearest neighbor sampling
+#' rarefaction(inv_comm, method='spat', xy_coords=inv_plot_attr[ , c('x','y')])
+rarefaction = function(x, method, effort=NULL, xy_coords=NULL) {
+    if (!any(method %in% c('indiv', 'samp', 'spat')))
         stop('method must be "indiv" or "samp" for individual or sample based rarefaction, respectively')
```
Comment (dmcglinn) [inline R/mobr.R]: good catch I'll implement this

File: R/mobr.R [hunk 4]
```diff
@@ -57,290 +59,218 @@ make_comm_obj = function(x, plot_attr, binary=FALSE) {
...
+#' within and between plot spatial aggregation, but maintains species group
+#' abundance and therefore observed group N, and 2) 'swapN': the total number of
+#' individuals in a plot is shuffled and then that many individuals are drawn
+#' randomly from the group specific species-abundance distribution for each plot
+#' which provides a means of removing group differenes in the total number of
+#' individuals.
 #' 
-#' @param comm a site-by-species matrix
-#' @param swap either 'noagg' or 'swapN' 
+#' @param comm community matrix with plots as rows and species columns.
```
Comment (dmcglinn) [inline R/mobr.R]: yea the number of individuals of a given species can change with `swapN` but not with `noagg`. I can add that in as a note.

File: R/mobr.R (hunk 2 above)
Comment (rueuntal) [inline R/mobr.R]: So should `coords` be removed from input, or is it sth else?

File: R/mobr.R (hunk 1 above)
Comment (dmcglinn) [inline R/mobr.R]: Agreed - in general the functions are not very robust right now to users that aren't paying attention to what they supply our functions. If time allows adding argument checks would be a good thing to do. More importantly though we should implement some formal tests to go along with the package and to aid future development.

**Pull request merged**